apps/client/src/app/services/import-activities.service.ts

```diff
@@ -202,16 +216,42 @@ export class ImportActivitiesService {
   }) {
     item = this.lowercaseKeys(item);
 
+    // Attempt to find currency in the CSV
     for (const key of ImportActivitiesService.CURRENCY_KEYS) {
       if (item[key]) {
         return item[key];
       }
     }
 
-    throw {
-      activities: content,
-      message: `activities.${index}.currency is not valid`
-    };
+    // If no currency in CSV, make an API request to get symbol data
+    const symbol = this.parseSymbol({ content, index, item });
+    const dataSource = this.parseDataSource({ item });
```

Comment:
Move up

Reply:
Hi could you please clarify move these lines up or move entire parseCurrency function up or rearrange it? Thanks

Reply:
Just switch the order:
```
const dataSource = this.parseDataSource({ item });
const symbol = this.parseSymbol({ content, index, item });
```